File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {
        int ans = 0;
        int n=colors.length;
        for (int i = 0; i < colors.length - 1; i++) {
           if(colors[i]!=colors[n-1]){
                ans=Math.max(ans,(n-1-i));
                break;
           }
        }
        for(int i=n-1;i>=1;i--){
            if(colors[0]!=colors[i]){
                ans=Math.max(ans,i);
                break;
            }
        }
        return ans;
    }
}

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        ans=0
        n=len(colors)
        for i in range(n-1,-1,-1):
            if colors[i]!=colors[0]:
                temp=abs(i)
                ans=max(ans,temp)
                break
        for i in range(n):
            if colors[i]!=colors[n-1]:
                temp=abs(n-i-1)
                ans=max(ans,temp)
                break
        return ans